examples/print-to-scale.js

```diff
@@ -0,0 +1,130 @@
+fetch(url)
+  .then(function(response) {
+    return response.text();
+  })
+  .then(function(text) {
+    const result = new WMTSCapabilities().read(text);
+    const options = optionsFromCapabilities(result, {
+      layer: 'OS_Open_Raster'
+    });
+    options.attributions = 'Contains OS data © Crown Copyright and database right ' + new Date().getFullYear();
+    options.crossOrigin = '';
+    options.projection = proj27700;
+    options.wrapX = false;
+    raster.setSource(new WMTS(options));
+  });
+
+
+const map = new Map({
+  layers: [raster],
+  controls: defaultControls({
+    attributionOptions: {collapsible: false}
+  }),
+  target: 'map',
+  view: new View({
+    center: [373500, 436500],
+    projection: proj27700,
+    zoom: 7
+  })
+});
+
+const scaleLine = new ScaleLine({bar: true, text: true});
+map.addControl(scaleLine);
+
+
+const dims = {
+  a0: [1189, 841],
+  a1: [841, 594],
+  a2: [594, 420],
+  a3: [420, 297],
+  a4: [297, 210],
+  a5: [210, 148]
+};
+
+
+// export options for html-to-image.
+// See: https://github.com/bubkoo/html-to-image#options
+const exportOptions = {
+  filter: function(element) {
+    const className = element.className || '';
+    return (
+      className.indexOf('ol-control') === -1 ||
+      className.indexOf('ol-scale') > -1 ||
+      (className.indexOf('ol-attribution') > -1 &&
+        className.indexOf('ol-uncollapsible'))
+    );
+  }
+};
+
+const exportButton = document.getElementById('export-pdf');
+
+exportButton.addEventListener('click', function() {
+
+  exportButton.disabled = true;
+  document.body.style.cursor = 'progress';
+
+  const format = document.getElementById('format').value;
+  const resolution = document.getElementById('resolution').value;
+  const scale = document.getElementById('scale').value;
+  const dim = dims[format];
+  const width = Math.round(dim[0] * resolution / 25.4);
+  const height = Math.round(dim[1] * resolution / 25.4);
+  const viewResolution = map.getView().getResolution();
+  const scaleResolution = scale / getPointResolution(
+    map.getView().getProjection(),
+    resolution / 25.4,
+    map.getView().getCenter()
+  );
+
+  map.once('rendercomplete', function() {
+    exportOptions.width = width;
+    exportOptions.height = height;
+    domtoimage.toJpeg(map.getViewport(), exportOptions).then(function(dataUrl) {
+      const pdf = new jsPDF('landscape', undefined, format);
+      pdf.addImage(dataUrl, 'JPEG', 0, 0, dim[0], dim[1]);
+      pdf.save('map.pdf');
+      // Reset original map size
+      scaleLine.setDpi();
+      scaleLine.setMinWidth();
+      map.getTargetElement().style.width = '';
+      map.getTargetElement().style.height = '';
+      map.updateSize();
+      map.getView().setResolution(viewResolution);
+      exportButton.disabled = false;
+      document.body.style.cursor = 'auto';
+    });
+  });
+
+  // Set print size
+  scaleLine.setDpi(resolution);
+  scaleLine.setMinWidth(resolution * 4 / 2.54); // 4cm
```

Comment:
Wouldn't it make sense to adjust widths automatically, so their units change from "pixels" to "pixels at screen resolution (25.4/28 dpi assumed)"? Then users would not have to worry about changing the minWidth when changing the dpi, and there would not be a need to add a `setMinWidth` method.